nms/app/packages/magmalte/app/components/cwf/CWFGateways.js

```diff
@@ -121,9 +128,18 @@ function CWFGateways(props: WithAlert & {}) {
     lastFetchTime,
   );
 
+  const {
+    isLoading: isHaPairsLoading,
+  } = useMagmaAPI(
+      MagmaV1API.getCwfByNetworkIdHaPairs,
+      {networkId},
+      useCallback(response => setHaPairs(map(response, h => h)), []),
+      lastFetchTime,
+  );
+
   useInterval(() => setLastFetchTime(Date.now()), REFRESH_INTERVAL);
 
-  if (!gateways || isLoading) {
+  if (!gateways || isLoading || !haPairs || isHaPairsLoading) {
```

Comment:
isLoading is misleading here. In the useMagmaApi as shown below, isLoading is set prior to invoking the callback. I would suggest getting rid of isLoading since you have the null check anyways or else remove the null check and isLoading and set a isLoading flag to false in the callback. 
```
  useEffect(() => {
    func
      .bind(MagmaV1API)((JSON.parse(jsonParams): TParams))
      .then(res => {
        setResponse(res);
        setError(null);
        setIsLoading(false); <===
        onResponse && onResponse(res);
      })
      .catch(err => {
        setError(err);
        setResponse(null);
        setIsLoading(false);
        if (err?.response?.status === 503) {
          enqueueSnackbar(
            'There was a problem connecting to the Orchestrator server',
            {variant: 'error'},
          );
        }
      });
  }, [jsonParams, func, cacheCounter, onResponse, enqueueSnackbar]);

  return {error, response, isLoading};```